### PR TITLE
Allow the use of BESU_CONFIG_FILE env var

### PR DESCRIPTION
**Pull request by antonydenyer**
## PR description
Enable the use of environment variable `BESU_CONFIG_FILE` to specify config toml

## Fixed Issue(s)
https://github.com/hyperledger/besu/issues/2455

## Changelog

- [x] I thought about the changelog and included a [changelog update if required](https://wiki.hyperledger.org/display/BESU/Changelog).

### CODE REVIEW

File: besu/src/main/java/org/hyperledger/besu/cli/util/ConfigOptionSearchAndRunHandler.java [hunk 1]
```diff
@@ -67,6 +67,16 @@ public ConfigOptionSearchAndRunHandler(
       } catch (final Exception e) {
         throw new ExecutionException(commandLine, e.getMessage(), e);
       }
+    } else if (environment.containsKey("BESU_CONFIG_FILE")) {
+      File toml = new File(environment.get("BESU_CONFIG_FILE"));
+      if (!toml.exists()) {
+        throw new ExecutionException(
+            commandLine,
+            String.format(
+                "toml file %s specified in environment variable BESU_CONFIG_FILE not found",
```
Comment (macfarla) [inline besu/src/main/java/org/hyperledger/besu/cli/util/ConfigOptionSearchAndRunHandler.java:76]: s/toml/TOML

File: besu/src/main/java/org/hyperledger/besu/cli/util/ConfigOptionSearchAndRunHandler.java [hunk 2]
```diff
@@ -67,6 +67,16 @@ public ConfigOptionSearchAndRunHandler(
       } catch (final Exception e) {
         throw new ExecutionException(commandLine, e.getMessage(), e);
       }
+    } else if (environment.containsKey("BESU_CONFIG_FILE")) {
+      File toml = new File(environment.get("BESU_CONFIG_FILE"));
```
Comment (macfarla) [inline besu/src/main/java/org/hyperledger/besu/cli/util/ConfigOptionSearchAndRunHandler.java:71]: final

**Review by macfarla — COMMENTED** [review]
couple of comments. I've tested this locally with success

File: besu/src/main/java/org/hyperledger/besu/cli/BesuCommand.java [hunk 3]
```diff
@@ -1342,8 +1342,7 @@ private void parse(
     // and eventually it will run regular parsing of the remaining options.
 
     final ConfigOptionSearchAndRunHandler configParsingHandler =
-        new ConfigOptionSearchAndRunHandler(
-            resultHandler, exceptionHandler, CONFIG_FILE_OPTION_NAME, environment);
+        new ConfigOptionSearchAndRunHandler(resultHandler, exceptionHandler, environment);
```
Comment (antonydenyer) [inline besu/src/main/java/org/hyperledger/besu/cli/BesuCommand.java:1346]: @macfarla got rid of injecting the config option name, felt like it just added confusion.

File: besu/src/main/java/org/hyperledger/besu/cli/util/ConfigOptionSearchAndRunHandler.java [hunk 4]
```diff
@@ -60,13 +57,23 @@ public ConfigOptionSearchAndRunHandler(
...
+      final OptionSpec configFileOption = parseResult.matchedOption("--config-file");
       try {
         return Optional.of(configFileOption.getter().get());
       } catch (final Exception e) {
         throw new ExecutionException(commandLine, e.getMessage(), e);
       }
+    } else if (environment.containsKey("BESU_CONFIG_FILE")) {
+      final File toml = new File(environment.get("BESU_CONFIG_FILE"));
+      if (!toml.exists()) {
+        throw new ExecutionException(
```
Comment (macfarla) [inline besu/src/main/java/org/hyperledger/besu/cli/util/ConfigOptionSearchAndRunHandler.java:70]: can this be ParameterException? PicoCLI makes the output nicer then

File: besu/src/main/java/org/hyperledger/besu/cli/util/ConfigOptionSearchAndRunHandler.java [hunk 5]
```diff
@@ -60,13 +57,23 @@ public ConfigOptionSearchAndRunHandler(
...
-    if (parseResult.hasMatchedOption(configFileOptionName)) {
-      final OptionSpec configFileOption = parseResult.matchedOption(configFileOptionName);
+    if (parseResult.hasMatchedOption("--config-file")) {
+      final OptionSpec configFileOption = parseResult.matchedOption("--config-file");
       try {
         return Optional.of(configFileOption.getter().get());
       } catch (final Exception e) {
         throw new ExecutionException(commandLine, e.getMessage(), e);
       }
+    } else if (environment.containsKey("BESU_CONFIG_FILE")) {
```
Comment (macfarla) [inline besu/src/main/java/org/hyperledger/besu/cli/util/ConfigOptionSearchAndRunHandler.java:67]: so if I specify --config-file AND the environment variable, the environment variable is ignored. Is this what we want? And if so, do we at least want a warning message for the user?

File: besu/src/main/java/org/hyperledger/besu/cli/BesuCommand.java (hunk 3 above)
Comment (macfarla) [reply besu/src/main/java/org/hyperledger/besu/cli/BesuCommand.java:1346]: 👍

File: besu/src/main/java/org/hyperledger/besu/cli/util/ConfigOptionSearchAndRunHandler.java (hunk 5 above)
Comment (antonydenyer) [reply besu/src/main/java/org/hyperledger/besu/cli/util/ConfigOptionSearchAndRunHandler.java:74]: Good spot! Throwing a `ParameterException` instead.

**Review by macfarla — APPROVED** [review]

**Pull request merged**